.circleci/config.yml

```diff
@@ -907,10 +907,47 @@ jobs:
       source-folder:
         type: string
         default: ""
-      folder:
+      before-steps:
+        type: steps
+        default: []
+      <<: *Params
+    executor: << parameters.e >>
+    environment:
+      MB_EDITION: << parameters.edition >>
+      CYPRESS_GROUP:  << parameters.cypress-group >>
+      DISPLAY: ""
+    steps:
+      - attach-workspace
+      - run-on-change:
+          checksum: '{{ checksum ".BACKEND-CHECKSUMS" }}-{{ checksum ".FRONTEND-CHECKSUMS" }}-{{ checksum ".E2E-TESTS-CHECKSUMS" }}'
+          steps:
+            - run-yarn-command:
+                command-name: Run Cypress tests Visual
+                before-steps:
+                  - restore_cache:
+                      name: Restore cached uberjar built in previous step
+                      <<: *CacheKeyUberjar
+                  - steps: << parameters.before-steps >>
+                # Make `source-folder` parameters optional. Translates to: if `parameter` => run associated flag (`--folder`)
+                command: |
+                  run test-visual-no-build <<# parameters.source-folder >> --folder << parameters.source-folder >> <</ parameters.source-folder >>
+                after-steps:
+                  - store_artifacts:
+                      path: /home/circleci/metabase/metabase/cypress
+                  - store_test_results:
+                      path: cypress/results
```

Comment:
As you described, I get that you wanted to separate visual tests from the main e2e tests because we don't want to record percy runs using currents dashboard. But, duplicating the whole workflow to achieve that seems like an overkill, when we have conditional parameters and conditional logic available to us in general.

Reply:
I've tried a few conditional parameters before that but it seemed that it does not work well with `run-yarn-command`